core/app/models/spree/order_contents.rb

```diff
@@ -18,6 +18,11 @@ def remove(variant, quantity = 1, options = {})
       after_add_or_remove(line_item, options)
     end
 
+    def remove_line_item(line_item, options = {})
+      line_item.destroy!
```

Comment:
@wuboy0307 I think we should call destroy using association here, `order.line_items.destroy(line_item)`. So if this association is used further in this call it would return updated set of line items.